vampyr/operators.cpp

```diff
@@ -0,0 +1,178 @@
+
+#include "PyRepresentableFunction.h"
+
+#include "operators/ABGVOperator.h"
+#include "operators/ConvolutionOperator.h"
+#include "operators/DerivativeOperator.h"
+#include "operators/HelmholtzOperator.h"
+#include "operators/PoissonOperator.h"
+#include "operators/IdentityConvolution.h"
+#include "operators/PHOperator.h"
+#include "operators/BSOperator.h"
+#include "operators/GreensKernel.h"
+#include "operators/DerivativeKernel.h"
+#include "operators/IdentityKernel.h"
+#include "operators/MWOperator.h"
+#include "operators/HelmholtzKernel.h"
+#include "operators/OperatorState.h"
+#include "operators/OperatorStatistics.h"
+
+#include "treebuilders/add.h"
+#include "treebuilders/apply.h"
+#include "treebuilders/grid.h"
+#include "treebuilders/multiply.h"
+#include "treebuilders/project.h"
+
+#include "functions/GaussFunc.h"
+#include "functions/GaussPoly.h"
+#include "functions/Gaussian.h"
+#include "functions/Polynomial.h"
+#include "functions/RepresentableFunction.h"
+#include "functions/AnalyticFunction.h"
+#include "functions/BoysFunction.h"
+#include "functions/LegendrePoly.h"
+
+
+
+using namespace mrcpp;
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace vampyr {
+void operators(py::module &m) {
+    const auto D = 3;
+
+
+ 	  m.def(
+        "divergence",
+        py::overload_cast<FunctionTree<D> &, DerivativeOperator<D> &, FunctionTreeVector<D> &>(&mrcpp::divergence<D>));
+
+    m.def("gradient", py::overload_cast<DerivativeOperator<D> &, FunctionTree<D> &>(&gradient<D>));
+ 
+   //ABGVOperator
+    py::class_<ABGVOperator<D>> abgvoperator(m, "ABGVOperator");
+    abgvoperator.def(py::init<MultiResolutionAnalysis<D> &, double, double>(),
+             "Derivative Operator: The ABGVOperator for differentiation");
+
+    //BSOperator
+ 	  py::class_<BSOperator<D>> bsoperator(m, "BSOperator");
+    bsoperator.def(py::init<MultiResolutionAnalysis<D> &, int>());
+
+    //ConvolutionOperator
+	  py::class_<ConvolutionOperator<D>> convop(m, "ConvolutionOperator");
+    convop.def(py::init<MultiResolutionAnalysis<D> &, double>());
+
+    //DerivativeKernel
+    py::class_<DerivativeKernel> derivativekernel(m, "DerivativeKernel");
+    derivativekernel.def(py::init<double>(),
+             "eps"_a,
+             "GreensKernel: DerivativeKernel");
+
+    //DerivativeOperator
+    py::class_<DerivativeOperator<D>> derivativeoperator(m, "Derivative Operator");
+    derivativeoperator.def(py::init<MultiResolutionAnalysis<D>>());
+
+    //GreensKernel
+	  py::class_<GreensKernel> greensKernel(m, "GreensKernel");
+
+    //HelmholtzKernel
+    py::class_<HelmholtzKernel>(m, "HelmholtzKernel", greensKernel)
+        .def(py::init<double, double, double, double>(),
+             "m"_a,
+             "eps"_a,
+             "r_min"_a,
+             "r_max"_a,
+             "GreensKernel: HelmholtzKernel");    
+
+    //HelmholtzOperator
+    py::class_<HelmholtzOperator>(m, "HelmholtzOperator", convop)
```

Comment:
This needs to be written in the same way as the `PoissonOperator` with a `multiple_inheritance`

Reply:
Have a look at the indentations